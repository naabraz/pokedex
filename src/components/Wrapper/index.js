// @flow

import React from 'react'

import {
  WrapperView,
  ContentView,
} from './style'

const IWrapper = {
  children: Object,
}

const Wrapper = (props: IWrapper) => (
  <WrapperView>
    <ContentView>
      {props.children}
    </ContentView>
  </WrapperView>
)

export default Wrapper
