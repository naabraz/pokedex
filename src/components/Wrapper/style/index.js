import styled from 'styled-components/native'

import {
  red,
  lightBlue,
} from 'themes'

export const WrapperView = styled.View`
  display: flex
  align-items: center
  justify-content: center
  background-color: ${red}
  height: 100%
`
export const ContentView = styled.View`
  background-color: ${lightBlue}
  align-items: center
  justify-content: center
  height: 85%
  width: 85%
  border-radius: 5
`
