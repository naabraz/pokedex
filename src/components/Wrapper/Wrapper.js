import React from 'react'
import PropTypes from 'prop-types'

import { WrapperView } from './style'

const Wrapper = props => (
  <WrapperView>
    {props.children}
  </WrapperView>
)

Wrapper.propTypes = {
  children: PropTypes.node,
}

export default Wrapper
