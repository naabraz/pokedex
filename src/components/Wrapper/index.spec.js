import React from 'react';
import renderer from 'react-test-renderer';

import Wrapper from '.';

describe('Wrapper component test', () => {
  it('Should render component correctly', () => {
    const wrapper = renderer.create(<Wrapper>This is a String</Wrapper>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
