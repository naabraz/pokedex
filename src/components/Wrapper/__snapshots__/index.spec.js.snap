// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper component test Should render component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#e32722",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#64afe1",
          "borderRadius": 5,
          "height": "85%",
          "justifyContent": "center",
          "width": "85%",
        },
      ]
    }
  >
    This is a String
  </View>
</View>
`;
